getting_started.ipynb

```diff
@@ -0,0 +1,1962 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab9705c6-0512-48b3-b117-8e1bc385cf86",
+   "metadata": {},
+   "source": [
+    "## Tools and Structured output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e62770fa-c0e4-45ad-8f3b-5fd64e835857",
+   "metadata": {},
+   "source": [
+    "Import the required packages. `BaseModel` is a parent class that all tools will inherit from, and `Field` is used to define all properties of the tool."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 73,
+   "id": "84527cdd-6d45-4d18-a3c4-887a8537c9a5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_core.pydantic_v1 import BaseModel, Field"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e7b68b06-ccd1-40f1-9201-9814de191254",
+   "metadata": {},
+   "source": [
+    "Tools are essentially classes that can be passed to a chosen model to influence or structure how the response should be formatted or generated.\n",
+    "\n",
+    "**For example:**\n",
+    "\n",
+    "- A Weather tool with a specific API call could be passed so the model knows to use this specific API for data retrieval.\n",
+    "- A City tool with fields like `population`, `size`, and `main_language` so the model can return any city-related queries with an object containing the corresponding filled fields.\n",
+    "- An Image tool with a `url` field to be returned when asked to search for an image containing a dog, with the field containing the URL of the image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 75,
+   "id": "05f0ff77-07bd-4041-9ab5-c2ba13ca644c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Optional\n",
+    "#adds the ability to make a field Optional [NULL]\n",
+    "\n",
+    "class Person(BaseModel):\n",
+    "    '''Information about a given person'''\n",
+    "\n",
+    "    name: str = Field(..., description=\"The name of a person\")\n",
+    "    alive: bool = Field(..., description=\"Thether the person is alive or not\")\n",
+    "    profession: str = Field(..., description=\"What the person does for work or professionally\")\n",
+    "    noteable_features: str = Field(..., description=\"Any noteworthy features/achievements about the person\")\n",
+    "    hobbies: Optional[str] = Field(description=\"Anything hobbies the person may have\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 76,
+   "id": "c5d9b7cc-d127-4f28-ab8f-829f236ef419",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Person(name='Kate Sheppard', alive=False, profession='Suffragist', noteable_features=\"Kate Sheppard was the most prominent member of the women's suffrage movement in New Zealand. She played a key role in making New Zealand the first country to grant women the right to vote in 1893.\", hobbies=None)"
```

Comment:
I don't think "suffragist" is actually a profession? :P

Reply:
> hobbies=None

ouch 😄

Reply:
chatgpt was having trouble figuring out what Kate Sheppard did for work so I changed the `Person` tool

```python
class Person(BaseModel):
    '''Information about a given person'''

    name: str = Field(..., description="The name of a person")
    alive: bool = Field(..., description="Whether the person is alive or not")
    place_of_birth: str = Field(..., description="Where the person was born")
    noteable_features: str = Field(..., description="Any noteworthy features/achievements about the person")
    hobbies: str = Field(..., description="Any hobbies the person may have")
    ```
    
    the output is
    
    ```
    Person(name='Kate Sheppard', alive=False, place_of_birth='Liverpool, England', noteable_features="Leader of the women's suffrage movement in New Zealand, instrumental in making New Zealand the first country to grant women the right to vote in 1893.", hobbies='Activism, writing, public speaking')
    ```